### ISSUE TITLE
Missing drive, with Position showing a location in the middle of the drive

### ISSUE THREAD

**Issue by khorton** [adriankumpf/teslamate#307]
**Describe the bug**
I have a quite short commute to work (3.8 miles, which typically takes 6.5 minutes).  I had read that "TeslaMate should notice within 60s that the vehicle is awake" (https://github.com/adriankumpf/teslamate/issues/295#issuecomment-562848986), so this morning I did not take any action to resume polling.  After arriving at work, I checked TeslaMate, and saw that the morning commute was not listed under Drives, but that Position showed a location roughly one minute after leaving the garage.  The implication seems to be that TeslaMate saw the vehcile was awake, polled once, then attempted to let the vehicle go to sleep.  This does not match my understanding of how it should work.  Should I reset my expectations, or have I tickled a bug?

**Expected behavior**
If TeslaMate notes the vehicle is awake and driving, continue polling.

**Relevant entries from the logs**
_You **have to** provide the logs of the teslamate container / application unless the bug is limited to Grafana only. To fetch the logs of the teslamte container run the following command:_ `docker-compose logs teslamate`

```
teslamate_1  | 05:46:50.944 car_id=1 [info] Charging / Complete / 5.34 kWh – 89 min
teslamate_1  | 05:46:50.944 car_id=1 [info] Start / :online
teslamate_1  | 05:56:59.174 car_id=1 [info] Suspending logging
teslamate_1  | 06:17:59.537 car_id=1 [info] Suspending logging
teslamate_1  | 06:39:01.252 car_id=1 [info] Charging / SOC: 90% / Home
teslamate_1  | 06:43:11.983 car_id=1 [info] Derived efficiency factor: 152.0 Wh/km (9x confirmed)
teslamate_1  | 06:43:11.983 car_id=1 [info] Charging / Complete / 0.10999999999999943 kWh – 4 min
teslamate_1  | 06:43:11.983 car_id=1 [info] Start / :online
teslamate_1  | 06:44:08.020 [error] TeslaApi.Error / 504 – %{"error" => "upstream_timeout", "error_description" => "", "response" => nil}
teslamate_1  | 06:44:08.020 car_id=1 [warn] Error / upstream timeout
teslamate_1  | 06:44:23.056 [error] TeslaApi.Error / 504 – %{"error" => "upstream_timeout", "error_description" => "", "response" => nil}
teslamate_1  | 06:44:23.056 car_id=1 [warn] Error / upstream timeout
teslamate_1  | 06:48:05.050 [error] TeslaApi.Error / 504 – %{"error" => "upstream_timeout", "error_description" => "", "response" => nil}
teslamate_1  | 06:48:05.050 car_id=1 [warn] Error / upstream timeout
teslamate_1  | 06:48:20.096 [error] TeslaApi.Error / 504 – %{"error" => "upstream_timeout", "error_description" => "", "response" => nil}
teslamate_1  | 06:48:20.096 car_id=1 [warn] Error / upstream timeout
teslamate_1  | 06:48:25.320 car_id=1 [info] Driving / Start
teslamate_1  | 06:48:37.878 [error] TeslaApi.Error / 504 – %{"error" => "upstream_timeout", "error_description" => "", "response" => nil}
teslamate_1  | 06:48:37.878 car_id=1 [warn] Error / upstream timeout
teslamate_1  | 06:48:56.132 [error] TeslaApi.Error / 504 – %{"error" => "upstream_timeout", "error_description" => "", "response" => nil}
teslamate_1  | 06:48:56.133 car_id=1 [warn] Error / upstream timeout
teslamate_1  | 06:49:14.057 [error] TeslaApi.Error / 504 – %{"error" => "upstream_timeout", "error_description" => "", "response" => nil}
teslamate_1  | 06:49:14.057 car_id=1 [warn] Error / upstream timeout
teslamate_1  | 06:49:32.028 [error] TeslaApi.Error / 504 – %{"error" => "upstream_timeout", "error_description" => "", "response" => nil}
teslamate_1  | 06:49:32.029 car_id=1 [warn] Error / upstream timeout
teslamate_1  | 06:49:37.093 [error] TeslaApi.Error / 408 – %{
teslamate_1  |   "error" => "vehicle unavailable: {:error=>\"vehicle unavailable:\"}",
teslamate_1  |   "error_description" => "",
teslamate_1  |   "response" => nil
teslamate_1  | }
teslamate_1  | 06:49:37.118 car_id=1 [warn] Vehicle went offline while driving
teslamate_1  | 06:49:42.156 [error] TeslaApi.Error / 408 – %{
teslamate_1  |   "error" => "vehicle unavailable: {:error=>\"vehicle unavailable:\"}",
teslamate_1  |   "error_description" => "",
teslamate_1  |   "response" => nil
teslamate_1  | }
teslamate_1  | 06:49:47.244 [error] TeslaApi.Error / 408 – %{
teslamate_1  |   "error" => "vehicle unavailable: {:error=>\"vehicle unavailable:\"}",
teslamate_1  |   "error_description" => "",
teslamate_1  |   "response" => nil
teslamate_1  | }
teslamate_1  | 06:49:52.359 [error] TeslaApi.Error / 408 – %{
teslamate_1  |   "error" => "vehicle unavailable: {:error=>\"vehicle unavailable:\"}",
teslamate_1  |   "error_description" => "",
teslamate_1  |   "response" => nil
teslamate_1  | }
teslamate_1  | 06:49:57.616 [error] TeslaApi.Error / 408 – %{
teslamate_1  |   "error" => "vehicle unavailable: {:error=>\"vehicle unavailable:\"}",
teslamate_1  |   "error_description" => "",
teslamate_1  |   "response" => nil
teslamate_1  | }
teslamate_1  | 06:50:02.862 [error] TeslaApi.Error / 408 – %{
teslamate_1  |   "error" => "vehicle unavailable: {:error=>\"vehicle unavailable:\"}",
teslamate_1  |   "error_description" => "",
teslamate_1  |   "response" => nil
teslamate_1  | }
teslamate_1  | 06:50:08.017 [error] TeslaApi.Error / 408 – %{
teslamate_1  |   "error" => "vehicle unavailable: {:error=>\"vehicle unavailable:\"}",
teslamate_1  |   "error_description" => "",
teslamate_1  |   "response" => nil
teslamate_1  | }
teslamate_1  | 06:50:13.115 [error] TeslaApi.Error / 408 – %{
teslamate_1  |   "error" => "vehicle unavailable: {:error=>\"vehicle unavailable:\"}",
teslamate_1  |   "error_description" => "",
teslamate_1  |   "response" => nil
teslamate_1  | }
teslamate_1  | 06:50:18.713 [error] TeslaApi.Error / 408 – %{
teslamate_1  |   "error" => "vehicle unavailable: {:error=>\"vehicle unavailable:\"}",
teslamate_1  |   "error_description" => "",
teslamate_1  |   "response" => nil
teslamate_1  | }
teslamate_1  | 06:50:23.984 [error] TeslaApi.Error / 408 – %{
teslamate_1  |   "error" => "vehicle unavailable: {:error=>\"vehicle unavailable:\"}",
teslamate_1  |   "error_description" => "",
teslamate_1  |   "response" => nil
teslamate_1  | }
teslamate_1  | 06:50:29.082 [error] TeslaApi.Error / 408 – %{
teslamate_1  |   "error" => "vehicle unavailable: {:error=>\"vehicle unavailable:\"}",
teslamate_1  |   "error_description" => "",
teslamate_1  |   "response" => nil
teslamate_1  | }
teslamate_1  | 06:50:34.174 [error] TeslaApi.Error / 408 – %{
teslamate_1  |   "error" => "vehicle unavailable: {:error=>\"vehicle unavailable:\"}",
teslamate_1  |   "error_description" => "",
teslamate_1  |   "response" => nil
teslamate_1  | }
teslamate_1  | 06:50:39.330 [error] TeslaApi.Error / 408 – %{
teslamate_1  |   "error" => "vehicle unavailable: {:error=>\"vehicle unavailable:\"}",
teslamate_1  |   "error_description" => "",
teslamate_1  |   "response" => nil
teslamate_1  | }
teslamate_1  | 06:50:44.422 [error] TeslaApi.Error / 408 – %{
teslamate_1  |   "error" => "vehicle unavailable: {:error=>\"vehicle unavailable:\"}",
teslamate_1  |   "error_description" => "",
teslamate_1  |   "response" => nil
teslamate_1  | }
teslamate_1  | 06:50:49.740 [error] TeslaApi.Error / 408 – %{
teslamate_1  |   "error" => "vehicle unavailable: {:error=>\"vehicle unavailable:\"}",
teslamate_1  |   "error_description" => "",
teslamate_1  |   "response" => nil
teslamate_1  | }
teslamate_1  | 06:50:54.830 [error] TeslaApi.Error / 408 – %{
teslamate_1  |   "error" => "vehicle unavailable: {:error=>\"vehicle unavailable:\"}",
teslamate_1  |   "error_description" => "",
teslamate_1  |   "response" => nil
teslamate_1  | }
teslamate_1  | 06:50:59.932 [error] TeslaApi.Error / 408 – %{
teslamate_1  |   "error" => "vehicle unavailable: {:error=>\"vehicle unavailable:\"}",
teslamate_1  |   "error_description" => "",
teslamate_1  |   "response" => nil
teslamate_1  | }
teslamate_1  | 06:51:30.124 [error] TeslaApi.Error / 408 – %{
teslamate_1  |   "error" => "vehicle unavailable: {:error=>\"vehicle unavailable:\"}",
teslamate_1  |   "error_description" => "",
teslamate_1  |   "response" => nil
teslamate_1  | }
teslamate_1  | 06:52:00.261 [error] TeslaApi.Error / 408 – %{
teslamate_1  |   "error" => "vehicle unavailable: {:error=>\"vehicle unavailable:\"}",
teslamate_1  |   "error_description" => "",
teslamate_1  |   "response" => nil
teslamate_1  | }
teslamate_1  | 06:52:30.358 [error] TeslaApi.Error / 408 – %{
teslamate_1  |   "error" => "vehicle unavailable: {:error=>\"vehicle unavailable:\"}",
teslamate_1  |   "error_description" => "",
teslamate_1  |   "response" => nil
teslamate_1  | }
teslamate_1  | 06:53:00.441 [error] TeslaApi.Error / 408 – %{
teslamate_1  |   "error" => "vehicle unavailable: {:error=>\"vehicle unavailable:\"}",
teslamate_1  |   "error_description" => "",
teslamate_1  |   "response" => nil
teslamate_1  | }
teslamate_1  | 06:53:30.606 [error] TeslaApi.Error / 408 – %{
teslamate_1  |   "error" => "vehicle unavailable: {:error=>\"vehicle unavailable:\"}",
teslamate_1  |   "error_description" => "",
teslamate_1  |   "response" => nil
teslamate_1  | }
teslamate_1  | 06:54:00.747 [error] TeslaApi.Error / 408 – %{
teslamate_1  |   "error" => "vehicle unavailable: {:error=>\"vehicle unavailable:\"}",
teslamate_1  |   "error_description" => "",
teslamate_1  |   "response" => nil
teslamate_1  | }
teslamate_1  | 06:54:30.870 [error] TeslaApi.Error / 408 – %{
teslamate_1  |   "error" => "vehicle unavailable: {:error=>\"vehicle unavailable:\"}",
teslamate_1  |   "error_description" => "",
teslamate_1  |   "response" => nil
teslamate_1  | }
teslamate_1  | 06:55:01.007 [error] TeslaApi.Error / 408 – %{
teslamate_1  |   "error" => "vehicle unavailable: {:error=>\"vehicle unavailable:\"}",
teslamate_1  |   "error_description" => "",
teslamate_1  |   "response" => nil
teslamate_1  | }
teslamate_1  | 06:55:31.137 [error] TeslaApi.Error / 408 – %{
teslamate_1  |   "error" => "vehicle unavailable: {:error=>\"vehicle unavailable:\"}",
teslamate_1  |   "error_description" => "",
teslamate_1  |   "response" => nil
teslamate_1  | }
teslamate_1  | 06:56:01.281 [error] TeslaApi.Error / 408 – %{
teslamate_1  |   "error" => "vehicle unavailable: {:error=>\"vehicle unavailable:\"}",
teslamate_1  |   "error_description" => "",
teslamate_1  |   "response" => nil
teslamate_1  | }
teslamate_1  | 06:56:31.527 [error] TeslaApi.Error / 408 – %{
teslamate_1  |   "error" => "vehicle unavailable: {:error=>\"vehicle unavailable:\"}",
teslamate_1  |   "error_description" => "",
teslamate_1  |   "response" => nil
teslamate_1  | }
teslamate_1  | 06:57:01.637 [error] TeslaApi.Error / 408 – %{
teslamate_1  |   "error" => "vehicle unavailable: {:error=>\"vehicle unavailable:\"}",
teslamate_1  |   "error_description" => "",
teslamate_1  |   "response" => nil
teslamate_1  | }
teslamate_1  | 06:57:31.802 [error] TeslaApi.Error / 408 – %{
teslamate_1  |   "error" => "vehicle unavailable: {:error=>\"vehicle unavailable:\"}",
teslamate_1  |   "error_description" => "",
teslamate_1  |   "response" => nil
teslamate_1  | }
teslamate_1  | 06:58:01.946 [error] TeslaApi.Error / 408 – %{
teslamate_1  |   "error" => "vehicle unavailable: {:error=>\"vehicle unavailable:\"}",
teslamate_1  |   "error_description" => "",
teslamate_1  |   "response" => nil
teslamate_1  | }
teslamate_1  | 06:58:32.018 [error] TeslaApi.Error / 408 – %{
teslamate_1  |   "error" => "vehicle unavailable: {:error=>\"vehicle unavailable:\"}",
teslamate_1  |   "error_description" => "",
teslamate_1  |   "response" => nil
teslamate_1  | }
teslamate_1  | 06:59:02.152 [error] TeslaApi.Error / 408 – %{
teslamate_1  |   "error" => "vehicle unavailable: {:error=>\"vehicle unavailable:\"}",
teslamate_1  |   "error_description" => "",
teslamate_1  |   "response" => nil
teslamate_1  | }
teslamate_1  | 06:59:32.308 [error] TeslaApi.Error / 408 – %{
teslamate_1  |   "error" => "vehicle unavailable: {:error=>\"vehicle unavailable:\"}",
teslamate_1  |   "error_description" => "",
teslamate_1  |   "response" => nil
teslamate_1  | }
teslamate_1  | 07:00:02.540 [error] TeslaApi.Error / 408 – %{
teslamate_1  |   "error" => "vehicle unavailable: {:error=>\"vehicle unavailable:\"}",
teslamate_1  |   "error_description" => "",
teslamate_1  |   "response" => nil
teslamate_1  | }
teslamate_1  | 07:00:32.727 [error] TeslaApi.Error / 408 – %{
teslamate_1  |   "error" => "vehicle unavailable: {:error=>\"vehicle unavailable:\"}",
teslamate_1  |   "error_description" => "",
teslamate_1  |   "response" => nil
teslamate_1  | }
teslamate_1  | 07:01:02.871 [error] TeslaApi.Error / 408 – %{
teslamate_1  |   "error" => "vehicle unavailable: {:error=>\"vehicle unavailable:\"}",
teslamate_1  |   "error_description" => "",
teslamate_1  |   "response" => nil
teslamate_1  | }
teslamate_1  | 07:01:33.049 [error] TeslaApi.Error / 408 – %{
teslamate_1  |   "error" => "vehicle unavailable: {:error=>\"vehicle unavailable:\"}",
teslamate_1  |   "error_description" => "",
teslamate_1  |   "response" => nil
teslamate_1  | }
teslamate_1  | 07:02:03.142 [error] TeslaApi.Error / 408 – %{
teslamate_1  |   "error" => "vehicle unavailable: {:error=>\"vehicle unavailable:\"}",
teslamate_1  |   "error_description" => "",
teslamate_1  |   "response" => nil
teslamate_1  | }
teslamate_1  | 07:02:33.268 [error] TeslaApi.Error / 408 – %{
teslamate_1  |   "error" => "vehicle unavailable: {:error=>\"vehicle unavailable:\"}",
teslamate_1  |   "error_description" => "",
teslamate_1  |   "response" => nil
teslamate_1  | }
teslamate_1  | 07:03:03.415 [error] TeslaApi.Error / 408 – %{
teslamate_1  |   "error" => "vehicle unavailable: {:error=>\"vehicle unavailable:\"}",
teslamate_1  |   "error_description" => "",
teslamate_1  |   "response" => nil
teslamate_1  | }
teslamate_1  | 07:03:23.573 [info] GET /
teslamate_1  | 07:03:23.575 [info] Sent 200 in 2ms
teslamate_1  | 07:03:33.498 [error] TeslaApi.Error / 408 – %{
teslamate_1  |   "error" => "vehicle unavailable: {:error=>\"vehicle unavailable:\"}",
teslamate_1  |   "error_description" => "",
teslamate_1  |   "response" => nil
teslamate_1  | }
teslamate_1  | 07:04:03.661 [error] TeslaApi.Error / 408 – %{
teslamate_1  |   "error" => "vehicle unavailable: {:error=>\"vehicle unavailable:\"}",
teslamate_1  |   "error_description" => "",
teslamate_1  |   "response" => nil
teslamate_1  | }
teslamate_1  | 07:04:33.826 [error] TeslaApi.Error / 408 – %{
teslamate_1  |   "error" => "vehicle unavailable: {:error=>\"vehicle unavailable:\"}",
teslamate_1  |   "error_description" => "",
teslamate_1  |   "response" => nil
teslamate_1  | }
teslamate_1  | 07:04:35.613 car_id=1 [info] Driving / Timeout / 1 km – 1 min
teslamate_1  | 07:05:05.693 [error] TeslaApi.Error / 408 – %{
teslamate_1  |   "error" => "vehicle unavailable: {:error=>\"vehicle unavailable:\"}",
teslamate_1  |   "error_description" => "",
teslamate_1  |   "response" => nil
teslamate_1  | }
teslamate_1  | 07:05:36.085 [error] TeslaApi.Error / 408 – %{
teslamate_1  |   "error" => "vehicle unavailable: {:error=>\"vehicle unavailable:\"}",
teslamate_1  |   "error_description" => "",
teslamate_1  |   "response" => nil
teslamate_1  | }
teslamate_1  | 07:06:06.204 [error] TeslaApi.Error / 408 – %{
teslamate_1  |   "error" => "vehicle unavailable: {:error=>\"vehicle unavailable:\"}",
teslamate_1  |   "error_description" => "",
teslamate_1  |   "response" => nil
teslamate_1  | }
teslamate_1  | 07:06:06.344 [error] GenStateMachine :"1" terminating
teslamate_1  | ** (FunctionClauseError) no function clause matching in TeslaMate.Vehicles.Vehicle.handle_event/4
teslamate_1  |     (teslamate) lib/teslamate/vehicles/vehicle.ex:143: TeslaMate.Vehicles.Vehicle.handle_event(:internal, {:update, :asleep}, {:driving, {:offline, %TeslaApi.Vehicle{api_version: 7, backseat_token: nil, backseat_token_updated_at: nil, calendar_enabled: true, charge_state: %TeslaApi.Vehicle.State.Charge{trip_charging: false, battery_heater_on: false, conn_charge_cable: "<invalid>", charger_actual_current: 0, max_range_charge_counter: 0, user_charge_enable_request: nil, charge_port_cold_weather_mode: true, managed_charging_start_time: nil, charge_port_door_open: false, charger_voltage: 2, charger_phases: nil, charge_energy_added: 5.55, fast_charger_present: false, ideal_battery_range: 269.56, charge_miles_added_ideal: 22.5, timestamp: 1576162159276, charge_current_request: 48, managed_charging_user_canceled: false, usable_battery_level: 89, fast_charger_brand: "<invalid>", charge_limit_soc_std: 90, charger_power: 0, charge_to_max_range: false, charge_limit_soc_min: 50, battery_range: 269.56, est_battery_range: 165.79, charger_pilot_current: 48, time_to_full_charge: 0.0, charging_state: "Disconnected", charge_current_request_max: 48, not_enough_power_to_heat: nil, managed_charging_active: false, charge_port_latch: "Disengaged", fast_charger_type: "<invalid>", scheduled_charging_start_time: nil, charge_limit_soc: 90, battery_level: 90, charge_miles_added_rated: 22.5, charge_enable_request: false, charge_limit_soc_max: 100, charge_rate: 0.0, ...}, climate_state: %TeslaApi.Vehicle.State.Climate{battery_heater: false, battery_heater_no_power: nil, driver_temp_setting: 23.3, fan_status: 2, inside_temp: 23.5, is_auto_conditioning_on: true, is_climate_on: true, is_front_defroster_on: false, is_preconditioning: false, is_rear_defroster_on: false, left_temp_direction: 171, max_avail_temp: 28.0, min_avail_temp: 15.0, outside_temp: 4.5, passenger_temp_setting: 23.3, remote_heater_control_enabled: true, right_temp_direction: 171, seat_heater_left: 3, seat_heater_rear_center: 0, seat_heater_rear_left: 0, seat_heater_rear_right: 0, seat_heater_right: 0, side_mirror_heaters: false, smart_preconditioning: nil, timestamp: 1576162159276, wiper_blade_heater: false}, color: nil, display_name: "BlueBird", drive_state: %TeslaApi.Vehicle.State.Drive{gps_as_of: 1576162157, heading: 68, latitude: 47.187636, longitude: -119.373604, native_latitude: 47.187636, native_location_supported: 1, native_longitude: -119.373604, native_type: "wgs", power: -24, shift_state: "D", speed: 22, timestamp: 1576162159276}, gui_settings: %TeslaApi.Vehicle.State.Gui{gui_24_hour_time: true, gui_charge_rate_units: "kW", gui_distance_units: "mi/hr", gui_range_display: "Rated", gui_temperature_units: "F", timestamp: 1576162159276}, id: 41846229032906895, in_service: false, option_codes: ["AD15", "MDL3", "PBSB", "RENA", "BT37", "ID3W", "RF3G", "S3PB", "DRLH", "DV2W", "W39B", "APF0", "COUS", "BC3B", "CH07", "PC30", "FC3P", "FG31", "GLFR", "HL31", "HM31", "IL31", "LTPB", "MR31", "FM3B", "RS3H", "SA3P", "STCP", "SC04", "SU3C", "T3CA", "TW00", "TM00", ...], state: "online", tokens: ["961b7ff1f613ef89", "1ca93f9bd6751f41"], vehicle_config: %TeslaApi.Vehicle.State.VehicleConfig{can_accept_navigation_requests: true, can_actuate_trunks: true, car_special_type: "base", car_type: "model3", charge_port_type: "US", eu_vehicle: false, exterior_color: "DeepBlue", has_air_suspension: false, has_ludicrous_mode: false, key_version: 2, motorized_charge_port: true, perf_config: nil, plg: false, rear_seat_heaters: 1, rear_seat_type: nil, rhd: false, roof_color: "Glass", seat_type: nil, spoiler_type: "None", sun_roof_installed: nil, third_row_seats: "<invalid>", timestamp: 1576162159276, trim_badging: nil, wheel_type: "Pinwheel18"}, vehicle_id: 1275706294, vehicle_state: %TeslaApi.Vehicle.State.VehicleState{remote_start_enabled: true, vehicle_name: "BlueBird", autopark_state_v3: "standby", api_version: 7, parsed_calendar_supported: true, center_display_state: 4, sun_roof_percent_open: nil, dr: 0, software_update: %TeslaApi.Vehicle.State.VehicleState.SoftwareUpdate{expected_duration_sec: 2700, status: ""}, timestamp: 1576162159276, fp_window: 0, autopark_style: "dead_man", speed_limit_mode: %TeslaApi.Vehicle.State.VehicleState.SpeedLimit{active: false, current_limit_mph: 85.0, max_limit_mph: 90, min_limit_mph: 50, pin_code_set: false}, ft: 0, is_user_present: true, car_version: "2019.40.2.1 38f55d9f9205", media_state: %TeslaApi.Vehicle.State.VehicleState.MediaState{remote_control_enabled: true}, odometer: 5379.492769, locked: true, sentry_mode: false, pf: 0, valet_pin_needed: true, notifications_supported: true, rt: 0, df: 0, pr: 0, homelink_nearby: nil, rp_window: 0, ...}, vin: "5YJ3E1EB1KF452624"}}, nil}, %TeslaMate.Vehicles.Vehicle{car: %TeslaMate.Log.Car{__meta__: #Ecto.Schema.Metadata<:loaded, "cars">, charging_processes: #Ecto.Association.NotLoaded<association :charging_processes is not loaded>, drives: #Ecto.Association.NotLoaded<association :drives is not loaded>, efficiency: 0.152, eid: 41846229032906895, id: 1, inserted_at: ~N[2019-11-30 22:11:27], model: "3", name: "BlueBird", positions: #Ecto.Association.NotLoaded<association :positions is not loaded>, settings: %TeslaMate.Settings.CarSettings{__meta__: #Ecto.Schema.Metadata<:loaded, "car_settings">, car: #Ecto.Association.NotLoaded<association :car is not loaded>, id: 1, req_no_shift_state_reading: false, req_no_temp_reading: false, req_not_unlocked: true, sleep_mode_enabled: true, suspend_after_idle_min: 10, suspend_min: 21}, settings_id: 1, trim_badging: nil, updated_at: ~N[2019-12-04 13:27:00], vid: 1275706294, vin: "5YJ3E1EB1KF452624"}, deps: %{api: TeslaMate.Api, locations: TeslaMate.Locations, log: TeslaMate.Log, pubsub: Phoenix.PubSub, settings: TeslaMate.Settings, vehicles: TeslaMate.Vehicles}, last_response: %TeslaApi.Vehicle{api_version: 7, backseat_token: nil, backseat_token_updated_at: nil, calendar_enabled: true, charge_state: %TeslaApi.Vehicle.State.Charge{trip_charging: false, battery_heater_on: false, conn_charge_cable: "<invalid>", charger_actual_current: 0, max_range_charge_counter: 0, user_charge_enable_request: nil, charge_port_cold_weather_mode: true, managed_charging_start_time: nil, charge_port_door_open: false, charger_voltage: 2, charger_phases: nil, charge_energy_added: 5.55, fast_charger_present: false, ideal_battery_range: 269.56, charge_miles_added_ideal: 22.5, timestamp: 1576162159276, charge_current_request: 48, managed_charging_user_canceled: false, usable_battery_level: 89, fast_charger_brand: "<invalid>", charge_limit_soc_std: 90, charger_power: 0, charge_to_max_range: false, charge_limit_soc_min: 50, battery_range: 269.56, est_battery_range: 165.79, charger_pilot_current: 48, time_to_full_charge: 0.0, charging_state: "Disconnected", charge_current_request_max: 48, not_enough_power_to_heat: nil, managed_charging_active: false, charge_port_latch: "Disengaged", fast_charger_type: "<invalid>", scheduled_charging_start_time: nil, charge_limit_soc: 90, battery_level: 90, charge_miles_added_rated: 22.5, charge_enable_request: false, charge_limit_soc_max: 100, charge_rate: 0.0, scheduled_charging_pending: false}, climate_state: %TeslaApi.Vehicle.State.Climate{battery_heater: false, battery_heater_no_power: nil, driver_temp_setting: 23.3, fan_status: 2, inside_temp: 23.5, is_auto_conditioning_on: true, is_climate_on: true, is_front_defroster_on: false, is_preconditioning: false, is_rear_defroster_on: false, left_temp_direction: 171, max_avail_temp: 28.0, min_avail_temp: 15.0, outside_temp: 4.5, passenger_temp_setting: 23.3, remote_heater_control_enabled: true, right_temp_direction: 171, seat_heater_left: 3, seat_heater_rear_center: 0, seat_heater_rear_left: 0, seat_heater_rear_right: 0, seat_heater_right: 0, side_mirror_heaters: false, smart_preconditioning: nil, timestamp: 1576162159276, wiper_blade_heater: false}, color: nil, display_name: "BlueBird", drive_state: %TeslaApi.Vehicle.State.Drive{gps_ (truncated)
teslamate_1  | 07:06:06.406 car_id=1 [info] Start / :asleep
teslamate_1  | 07:54:37.995 [info] GET /
teslamate_1  | 07:54:37.997 [info] Sent 200 in 2ms
teslamate_1  | 08:54:41.144 [info] Adding elevation to 2 positions ...
teslamate_1  | 10:24:22.943 [info] GET /
teslamate_1  | 10:24:22.946 [info] Sent 200 in 2ms
teslamate_1  | 16:40:13.183 [error] TeslaApi.Error / 502 – "<!DOCTYPE html>\n<html>\n<head>\n  <title>We're sorry, but something went wrong (500)</title>\n  <style type=\"text/css\">\n    body { background-color: #fff; color: #666; text-align: center; font-family: arial, sans-serif; }\n    div.dialog {\n      width: 25em;\n      padding: 0 4em;\n      margin: 4em auto 0 auto;\n      border: 1px solid #ccc;\n      border-right-color: #999;\n      border-bottom-color: #999;\n    }\n    h1 { font-size: 100%; color: #f00; line-height: 1.5em; }\n  </style>\n</head>\n\n<body>\n  <!-- This file lives in public/500.html -->\n  <div class=\"dialog\">\n    <h1>We're sorry, but something went wrong.</h1>\n  </div>\n</body>\n</html>\n"
```

**Screenshots**
No morning commute shown for 12 Dec 2019:

![IMG_0280](https://user-images.githubusercontent.com/77668/70769041-b7e8d280-1d1c-11ea-9d45-8f2b283cd045.jpg)

After arriving at Work and parking, Position shows a spot early in the commute, over 3.5 miles from the eventual parking spot:

![IMG_0281](https://user-images.githubusercontent.com/77668/70769076-cc2ccf80-1d1c-11ea-8e73-a34c9c31d9e5.jpg)

The commute would have been from approximately 06:47 to 06:53.  States shows one minute of Driving in that period:

![IMG_0279](https://user-images.githubusercontent.com/77668/70769183-55dc9d00-1d1d-11ea-9f1a-511c3ae67698.jpg)

**Operating environment (please complete the following information):**
 - OS: Ubuntu 19.10, running in Docker
 - Version: v1.13.2

**Comment by BabyYeggie** [comment]
I get 504 errors when my LTE is out of service or the car is not connected.

408 errors seem to occur when the car is offline. It usually happens to me when the car is in deep sleep.

I seem to have a missed a drive with no indication anything is wrong.

`14:14:17.855 car_id=1 [info] Suspending logging

15:14:19.532 car_id=1 [info] Driving / Start

15:15:39.440 car_id=1 [info] Driving / Ended / 1 km – 1 min

15:15:39.440 car_id=1 [info] Start / :online

15:22:06.697 car_id=1 [info] Driving / Start

15:51:45.077 car_id=1 [info] Driving / Ended / 14 km – 30 min

15:51:45.077 car_id=1 [info] Start / :online

16:07:13.460 car_id=1 [info] Suspending logging

17:07:24.048 car_id=1 [info] Start / :asleep`

**Comment by adriankumpf** [comment]
>  Should I reset my expectations, or have I tickled a bug?

Your expectations are not wrong at all but you're missing a detail. I condensed the log output to highlight the relevant lines:

```
teslamate_1  | 06:43:11.983 car_id=1 [info] Start / :online
teslamate_1  | 06:48:25.320 car_id=1 [info] Driving / Start
teslamate_1  | 07:04:35.613 car_id=1 [info] Driving / Timeout / 1 km – 1 min
teslamate_1  | 07:06:06.344 [error] ... Vehicle.handle_event(:internal, {:update, :asleep}, {:driving, ...
teslamate_1  | 07:06:06.406 car_id=1 [info] Start / :asleep
```

* At the beginning, the API returned a few 504 errors indicating that the vehicle could not be reached, e.g. because it had poor cellular reception or switching from WiFi to cellular phones took a while. 
* At 06:48 a request came through and the logger noticed that the car was driving. After that, most of the requests returned a 504 error.  
* From 06:49 all requests failed with status code 408, which generally occurs when the vehicle is offline (or when there are problems with Tesla servers). 
* At 07:04 the logger pointed out that the car had been offline too long.
* At 07:06, a request was finally successful, but it was indicated that the car was now asleep. This led to a situation where the logger still expected the car to drive, but the transition from "driving" to "sleeping" is not supported. For this reason, the logger process was restarted.

In summary, the drive could not be logged successfully because the car was not reachable. It probably makes sense to shorten the polling interval after receiving an error if it is known that the vehicle is driving. But otherwise I think that the situation was handled quite well under the given conditions.

**Comment by mboerma** [comment]
I had a similar issue. It skipped 5 kilometers between drives. I cross checked with Teslafi and their data showed the correct starting and end points. It might be of course that they apply some logic when data is not coming in. The car can not magically teleport from the place you left it to somewhere else, unless it was put on a truck ;-) 

The drive is in a well covered area in the west on the Netherlands. 

This in the log. I started driving again at 14:47. 

```
teslamate_1  | 14:44:24.086 car_id=1 [info] Suspending logging
teslamate_1  | 14:56:25.378 car_id=1 [info] Driving / Start
teslamate_1  | 15:23:44.936 car_id=1 [info] Driving / Ended / 36 km – 27 min

```
Teslamate looks great, btw. Thanks for all the hard work.

**Comment by adriankumpf** [comment]
@mboerma in your case polling was halted/suspended right before your started driving. You can tweak the sleep mode settings (increase "Idle Time Before Trying to Sleep") or use Bluetooth hints to prevent this. See [Sleep Mode Configuration](https://github.com/adriankumpf/teslamate/blob/master/docs/configuration/sleep.md#idle-time-before-trying-to-sleep) for more information.

**Comment by adriankumpf** [comment]
Closing this for now if there are no further questions.
